docs/index.bs

```diff
@@ -3175,6 +3235,78 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
       1. Return |response|.
   </section>
 
+  <section algorithm>
+    <h3 id="verify-router-rule-algorithm"><dfn>VerifyRouterRule</dfn></h3>
+
+      : Input
+      :: |rule|, a {{RouterRule}}
+      :: |serviceWorker|, a [=/service worker=]
+      : Output
+      :: a boolean
+
+      1. If |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"] is the empty string, return false.
+      1. Let |rawPattern| be |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"].
+      1. If |rawPattern| is a [=string=], then:
+        1. Let |baseURL| be |serviceWorker|'s [=service worker/script url=].
+        1. Set |pattern| to the result of constructing a {{URLPattern}} using the {{URLPattern/URLPattern(input, baseURL)}} constructor steps given |rawPattern| and |baseURL|. If those steps throw, catch the exception and return false.
+      1. Otherwise, if |rawPattern| is {{URLPatternInit}}:
+        1. Set |pattern| to the result of constructing a {{URLPattern}} using the {{URLPattern/URLPattern(input)}} constructor steps given |rawPattern|. If those steps throw, catch the exception and return false.
+      1. Otherwise, return false.
+      1. If running the [=VerifyURLPattern=] algorithm with |pattern| returns false, return false.
+      1. Set |rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"]'s associated {{URLPattern}} to |pattern|.
```

Comment:
As discussed in the previous PR, you cannot create an "associated URLPattern" concept for a string (or URLPatternCompatible), which is what `|rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"]` is.

You need some sort of parallel structure, representing the fully parsed routing rule. That will probably be a new [struct](https://infra.spec.whatwg.org/#struct) you define with two members: URL pattern (a `URLPattern`), and source (a `RouterSourceEnum`).

This algorithm will then be responsible for parsing the Web IDL dictionary into this struct, so that you can store it in the "static router rules object".

Reply:
Instead of associating the URLPattern, I tried to build them on-demand.
Chromium implementation just verifies the URLPattern, and do not reuse the built pattern here.

Reply:
I guess that works too; it feels inefficient and a bit silly, but since parsing gives the same result each time (and the base URL cannot change), it is OK.